### PR TITLE
DUOS-1274[risk=no] Refactored searchFilteredElectionList in MetricsService for correct % SRP Approved

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1274](https://broadworkbench.atlassian.net/browse/DUOS-1274)

PR corrects null check in searchFilteredElectionList and date references in associated MetricsDAO query.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/db/MetricsDAO.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
   @SqlQuery(
     "SELECT * from ( "
-      + "SELECT e.*, v.vote finalvote, v.rationale finalrationale, v.createdate finalvotedate, MAX(e.electionid) "
+      + "SELECT e.*, v.vote finalvote, v.rationale finalrationale, v.updatedate finalvotedate, MAX(e.electionid) "
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/MetricsDAO.java]: Looking at `VoteDAO`, the only time we set a create date is on the update method. On dev, there are many more records with null update dates than create dates:
```
consent=> select count(voteid) from vote;
 count 
-------
 16971
(1 row)

consent=> select count(voteid) from vote where createdate is null;
 count 
-------
 14182
(1 row)

consent=> select count(voteid) from vote where updatedate is null;
 count 
-------
 16489
(1 row)
```

In ElectionDAO, we use the createDate for the final vote date exclusively so I don't think this is the correct approach.

**Review by rushtong — CHANGES_REQUESTED** [review]
I'm not confident this will solve the problem - have you run this code against staging? Let's get that set up for you if you haven't.

**Comment by rushtong** [comment]
Re-reviewing, I was incorrectly looking at your LC branch. Looking at staging, this does indeed show more accurate numbers.

Screen Shot 2021-05-27 at 3 20 27 PM

**Review by rushtong — APPROVED** [review]
This does show more accurate numbers. It also introduces room for improving the vote population code which is lacking.

**Review by raejohanek — APPROVED** [review]
I'm not exactly show how to test this out since it was an issue on staging, but the code looks good and when I check it out locally everything works well!

File: src/main/java/org/broadinstitute/consent/http/db/MetricsDAO.java [hunk 2]
```diff
@@ -27,7 +27,12 @@
 
   @SqlQuery(
     "SELECT * from ( "
-      + "SELECT e.*, v.vote finalvote, v.rationale finalrationale, v.createdate finalvotedate, MAX(e.electionid) "
+      + "SELECT e.*, v.vote finalvote, "
+      + "CASE "
+        + "WHEN v.updatedate IS NULL THEN v.createdate "
+        + "ELSE v.updatedate "
+      + "END as finalvotedate, "
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/MetricsDAO.java:34]: Very, very nice solution!

**Review by rushtong — APPROVED** [review]
Nice work!

**Pull request merged**